tensorflow/compiler/mlir/hlo/lib/Dialect/lhlo/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -462,6 +462,96 @@ class GatherOpConverter : public OpRewritePattern<GatherOp> {
   }
 };
 
+/// Converts PadOp to Affine nest form.
+/// Pseudocode:
+///   1. Fill `output` tensor with `padding_value`.
+///   2. Compute AffineMap for store into `output`.
+///      out_idx = edge_padding_low +
+///                operand_idx * (1 + interior_padding)
+///   3. Create nested loop from `operand` shape.
+///      3.1 load from `operand`.
+///      3.2 store into `output`.
+/// Limitation for now:
+///   interior_padding == 0 && edge_padding_* >= 0
+struct PadOpConverter : public OpRewritePattern<PadOp> {
+  using OpRewritePattern<PadOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(PadOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value operand = op.operand();
+    Value padding_value = op.padding_value();
+    Value output = op.output();
+
+    ShapedType operand_type = operand.getType().dyn_cast<ShapedType>();
+    ShapedType output_type = output.getType().dyn_cast<ShapedType>();
+    unsigned rank = operand_type.getRank();
+    assert(output_type.getRank() == rank);
+    ArrayRef<int64_t> operand_shape = operand_type.getShape();
+    ArrayRef<int64_t> output_shape = output_type.getShape();
+
+    SmallVector<int64_t> edge_padding_low(
+        op.edge_padding_low().getValues<int64_t>().begin(),
+        op.edge_padding_low().getValues<int64_t>().end());
+    SmallVector<int64_t> edge_padding_high(
+        op.edge_padding_high().getValues<int64_t>().begin(),
+        op.edge_padding_high().getValues<int64_t>().end());
+    SmallVector<int64_t> interior_padding(
+        op.interior_padding().getValues<int64_t>().begin(),
+        op.interior_padding().getValues<int64_t>().end());
+
+    assert(edge_padding_low.size() == rank && edge_padding_high.size() == rank &&
+           interior_padding.size() == rank);
```

Comment:
Same question as above: is this covered by the verifier?

Reply:
Likewise. There is no verifier checks for `lmhlo.Pad` regarding `interior_padding` and `edge_padding_*` attributes.

Reply:
OK, I guess another question then may be: **should** there be a verifier for these?

Reply:
Yes, in my opinion, there should be a verifier implemented for the these.

As per : https://www.tensorflow.org/xla/operation_semantics#pad , it is required by padding configurations to specify the amount of edge padding and the interior padding for `each` dimension.

We don't have verifier for this both on `mhlo` and `lhlo`.

I can implement it for both and include it as part of this PR.

Let me know if we should do that OR get rid of these checks from this PR since they `should` be taken care by a verifier and later on raise a PR for verifiers.

Reply:
If you can introduce the verifier and remove the checks that would be perfect, thanks!

Reply:
Hi @joker-eph 

I've included a verifier for `lmhlo.Pad` and all this logic will now be taken care by the verifier itself.
I've included another verifier check for the padding configurations too which check for the expected output shape after paddings are applied.

You may re-review this now.